common/app/routes/challenges/redux/reducer.js

```diff
@@ -74,6 +75,7 @@ const mainReducer = handleActions(
       // used mainly to find code storage
       legacyKey: challenge.name,
       challenge: challenge.dashedName,
+      blockType: challenge.block,
```

Comment:
I could just add `challengeSpec.name` to the challenge object in `./seed/index.js`

It would save all the messing around with entities and blocks